src/pymor/algorithms/pod.py

```diff
@@ -64,7 +67,13 @@ def pod(A, product=None, modes=None, rtol=4e-8, atol=0., l2_err=0.,
 
     svd_va = method_of_snapshots if method == 'method_of_snapshots' else qr_svd
     with logger.block('Computing SVD ...'):
-        POD, SVALS, _ = svd_va(A, product=product, modes=modes, rtol=rtol, atol=atol,
-                               l2_err=l2_err, **svd_params)
+        POD, SVALS, _ = svd_va(A, product=product, modes=modes, rtol=rtol, atol=atol, l2_err=l2_err)
+
+    if np.isfinite(orth_tol):
+        logger.info('Checking orthonormality ...')
+        err = np.max(np.abs(POD.inner(POD, product) - np.eye(len(POD))))
+        if err >= orth_tol:
+            logger.info('Reorthogonalizing POD modes ...')
+            gram_schmidt(POD, product=product, copy=False, check=False)
```

Comment:
Why `check=False`. Maybe we should add a `orth_params` dict as a parameter to `pod` that is passed to `gram_schmidt`?

Reply:
I set `check=False` to reduce computation (my impression was you prefer to avoid this check as [here](https://github.com/pymor/pymor/pull/718#discussion_r342985359)). I'm also fine with `check=True`.

Reply:
If set `check=False` there is no way to ensure that the result of `pod` is actually orthonormal. In particular when working with a broken inner `product` this can be an important safety check, so I would like to keep the possibility to automatically perform it. So I would vote for either `check=True` or add a dict of options for `gram_schmidt` to the signature of `pod`.

Reply:
Ok. I'm still confused about having a check in `pod` but not in SVD methods (particularly `method_of_snapshots`)...

Reply:
I guess one could argue that orthognoality is more important when you want to use the left-singular vectors as a basis, and for that you would typically call `pod`. That being said, it would also be an option to also add the check to `method_of_snapshots` and disable it when being called from `pod`.

Reply:
I kind of like that `method_of_snapshots` and `qr_svd` have the same signature, although I would like for `method_of_snapshots` to guarantee orthonormality since it's an SVD method. For now, I removed `check=False` and added a warning to the docstring `method_of_snapshots`.